src/klein/_typing_compat.py

```diff
@@ -0,0 +1,12 @@
+from typing import TYPE_CHECKING
```

Comment:
Mabye _typing_compat is common knowledge.
Maybe it would help to have a docstring with the purpose of this module, to help decide whether to add more code here or search for another place to add typing related helpers

Reply:
Will update, it'll be good to have an additional commit to poke GHA